server/chat/backend/agent/tools/cloud_exec_tool.py

```
@@ -1096,6 +1095,45 @@ def execute_tailscale_command(command: str, isolated_env: dict) -> dict:
+        if not api_token:
+            logger.error("Missing Fly.io api_token")
+            return False, None, None, None
+
+        isolated_env = {
+            "PATH": os.environ.get("PATH", ""),
+            "HOME": _ISOLATED_HOME,
+            "USER": os.environ.get("USER", ""),
+            "FLY_API_TOKEN": api_token,
+        }
+
+        logger.info(f"Fly.io isolated environment configured (org: {org_slug})")
```

Ruff (0.15.15)

[warning] Logging statement uses f-string

(G004)